admin-writer/src/main/resources/raw_data/db/migration/R__1_Views.sql

```diff
@@ -52,31 +51,99 @@ CREATE OR REPLACE VIEW raw_data.view_methods_with_rules AS
+		methods.build_id,
+		instances.env_id,
+		builds.branch,
+		definitions.tags AS test_tags,
+        methods.signature,
+        methods.body_checksum,
+        BIT_LENGTH(SUBSTRING(coverage.probes FROM methods.probe_start_pos + 1 FOR methods.probes_count)) AS probes_count,
+        SUBSTRING(coverage.probes FROM methods.probe_start_pos + 1 FOR methods.probes_count) AS probes,
+        BIT_COUNT(SUBSTRING(coverage.probes FROM methods.probe_start_pos + 1 FOR methods.probes_count)) AS covered_probes,
+        coverage.created_at,
+        builds.created_at AS build_created_at
+    FROM raw_data.coverage coverage
+	JOIN raw_data.methods methods ON methods.classname = coverage.classname
+    JOIN raw_data.instances instances ON instances.id = coverage.instance_id
+    JOIN raw_data.builds builds ON builds.id = methods.build_id AND builds.id = instances.build_id
+    LEFT JOIN raw_data.test_launches launches ON launches.id = coverage.test_id
+	LEFT JOIN raw_data.test_definitions definitions ON definitions.id = launches.test_definition_id
+    WHERE TRUE
+	  AND methods.probes_count > 0
+	  AND BIT_COUNT(SUBSTRING(coverage.probes FROM methods.probe_start_pos + 1 FOR methods.probes_count)) > 0
+      AND BIT_LENGTH(SUBSTRING(coverage.probes FROM methods.probe_start_pos + 1 FOR methods.probes_count)) = methods.probes_count;
 
 -----------------------------------------------------------------
 
 -----------------------------------------------------------------
-CREATE OR REPLACE VIEW raw_data.view_build_coverage AS
-    WITH
-    CoverageGroupedByMethod AS (
-        SELECT
-          group_id,
-          app_id,
-          build_id,
-          env_id,
-          signature,
-          BIT_COUNT(BIT_OR(probes)) AS covered_probes
-        FROM raw_data.view_methods_coverage
-        GROUP BY group_id, app_id, build_id, env_id, signature
-    ),
-    CoverageGroupedByEnv AS (
-        SELECT
-          group_id,
-          app_id,
-          build_id,
-          env_id,
-          SUM(covered_probes) AS covered_probes
-        FROM CoverageGroupedByMethod
-        GROUP BY group_id, app_id, build_id, env_id
-    )
-    SELECT * FROM CoverageGroupedByEnv;
\ No newline at end of file
+CREATE OR REPLACE VIEW raw_data.view_test_sessions AS
+    SELECT
+        ts.id AS test_session_id,
+        ts.group_id,
+        ts.test_task_id,
+        ts.started_at,
+        COUNT(*) AS tests,
+        (CASE WHEN SUM(CASE WHEN tl.result = 'FAILED' THEN 1 ELSE 0 END) > 0 THEN 'FAILED' ELSE 'PASSED' END) AS result,
+        (
+            SELECT
+              i.build_id
+            FROM
+              raw_data.coverage c
+            JOIN raw_data.test_launches tl ON tl.id = c.test_id
+            JOIN raw_data.instances i ON i.id = c.instance_id
+            WHERE tl.test_session_id = ts.id
+            LIMIT 1
+          ) AS build_id,
+        sum(tl.duration) AS duration,
+        SUM(CASE WHEN tl.result = 'FAILED' THEN 1 ELSE 0 END) AS failed,
```

Comment:
Wwhy do we want multiple columns with booleans instead of a single field with actual status?